firmware/Core/Src/telecommands/telecommand_definitions.c

```diff
@@ -1046,3 +1056,194 @@ uint8_t TCMDEXEC_reboot(const char *args_str, TCMD_TelecommandChannel_enum_t tcm
     return 0;
 }
 
+/// @brief Send artibrary configuration commands to a peripheral (MPI, GPS, CAMERA)
+/// @param args_str 
+/// - Arg 0: UART port to send data to
+/// - Arg 1: Data to be sent (bytes specified in base64)
+/// @param tcmd_channel The channel on which the telecommand was received, and on which the response should be sent
+/// @param response_output_buf The buffer to write the response to
+/// @param response_output_buf_len The maximum length of the response_output_buf (its size)
+/// @return 0: Success, 1: Error parsing args, 2: Invalid uart port requested, 3: Error transmitting data, 4: Error receiving data
+uint8_t TCMDEXEC_peripheral_send_receive_data(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                        char *response_output_buf, uint16_t response_output_buf_len) {
+
+    // Parse UART port argument
+    char arg_uart_port_name[10] = "";
+    const uint8_t uart_port_name_parse_result = TCMD_extract_string_arg(args_str, 0, arg_uart_port_name, 10);
+
+    // Parse hex-encoded config command (string to bytes)
+    const uint16_t args_bytes_to_send_size = (strlen(args_str) - strlen(arg_uart_port_name))/2;             // Expected size of config command byte array
```

Comment:
This won't work as the hex value could be sent with/without spaces changing the actual size of the buffer, also, I think we want to send 5 Kb

Reply:
Thanks for pointing that out! I have updated this bug to now support 5KBs of fixed transmission buffer size and handle the with/without spaces scenario in the latest commit.